src/track/track.cpp

```diff
@@ -813,9 +815,17 @@ void Track::slotCueUpdated() {
     emit cuesUpdated();
 }
 
-CuePointer Track::createAndAddCue() {
+CuePointer Track::createAndAddCue(
+        mixxx::CueType type,
+        int hotCueIndex,
+        double sampleStartPosition,
+        double sampleEndPosition) {
     QMutexLocker lock(&m_qMutex);
```

Comment:
Move the lock down to the point when it is actually needed. We don't need to protect the global system allocator or the Qt connect functions.